clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -1,13 +1,38 @@
 package clusterdeployer
 
 import (
+	"fmt"
+
 	"sigs.k8s.io/cluster-api/errors"
 	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
+// Can provision a kubernetes cluster
+type ClusterProvisioner interface {
+	Create() error
+	Delete() error
+	GetKubeconfig() (string, error)
+}
+
 type ClusterDeployer struct {
+	externalProvisioner    ClusterProvisioner
+	cleanupExternalCluster bool
+}
+
+func New(externalProvisioner ClusterProvisioner, cleanupExternalCluster bool) *ClusterDeployer {
+	return &ClusterDeployer{
+		externalProvisioner:    externalProvisioner,
+		cleanupExternalCluster: cleanupExternalCluster,
+	}
 }
 
-func (*ClusterDeployer) Create(_ *clusterv1.Cluster, _ []*clusterv1.Machine) error {
-	return errors.NotImplementedError
+// Creates the a cluster from the provided cluster definition and machine list.
+func (d *ClusterDeployer) Create(_ *clusterv1.Cluster, _ []*clusterv1.Machine) error {
+	if err := d.externalProvisioner.Create(); err != nil {
```

Comment:
I assume the external cluster will eventually run the machine and cluster controllers too, correct? Will that happen in this create function, or would that live somewhere else?

Reply:
That would live in this create function in the next PR. :)